market_env/env.py

```diff
@@ -79,16 +98,37 @@ def build_block(self, block_id) -> "Block":
         block = Block(block_id)
         for transaction in ordered_transactions:
             block.add_transaction(transaction)
+
+        state = self.env  # Assuming env is an instance of ChainEnv
+        if state.chosen_builder_index != -1:
+            if state.chosen_builder_index == self.builder_id:
+                intermediate_block = IntermediateBlock(...)
+                self.publish_intermediate_block(intermediate_block)
+
         return block
 
-    # order transactions by gas usage
-    def order_transactions(
-        self, transactions: List["Transaction"]
+    # select which transactions to include in the block base on gas fee and cannot exceed gas limit.
+    # sorts the transactions based on gas fee in descending order and selects transactions until the gas limit is reached.
+    def select_transactions(
+        self, transactions: List["Transaction"], gas_limit: int
```

Comment:
For 3.9+, you can use list[Transactions] without having to import List type @TammyQAQ